src/stanc/stanc.ml

```diff
@@ -359,7 +360,11 @@ let use_file filename =
         if !no_soa_opt then {base_optims with optimize_soa= false}
         else if !soa_opt then {base_optims with optimize_soa= true}
         else base_optims in
-      Optimize.optimization_suite ~settings:set_optims tx_mir in
+      let set_optims_opencl =
+        if !opencl_opt then
+          {set_optims with optimize_opencl= true; optimize_soa= false}
```

Comment:
If `opencl` and `soa` are mutually exclusive, I'd rather enforce this inside the optimization module than requiring the settings to keep it correct (you can just have the soa pass run iff `optimize_soa && not optimize_opencl`)

Reply:
Yes, though do we need to still use this for O1? Maybe not actually